resource/resource.go

```diff
@@ -409,3 +423,18 @@ func GitCommit(repoPath string) (string, error) {
 
 	return sha, nil
 }
+
+func buildInstanceVarsURL(instanceVarStr string) (string, error) {
+	var varsMap map[string]interface{}
+	err := json.Unmarshal([]byte(instanceVarStr), &varsMap)
+	if err != nil {
+		return "", err
+	}
+	var vars []string
+	for k, v := range varsMap {
+		vars = append(vars, fmt.Sprintf("vars.%s=\"%s\"",
+			url.QueryEscape(k), url.QueryEscape(v.(string))))
```

Comment:
Hey, sorry for the random comment 😁 , but stumbled upon this PR and figured I'd take a look (since I worked on a lot of the instance vars stuff in Concourse).

This won't work in general, since instance vars are allowed to be arbitrary JSON  (not necessarily strings). Additionally, if instance vars are JSON objects, we use dot-notation to represent accessing subfields - for instance, if you have instance vars:

```json
{
  "foo": {"bar": 123}
}
```

that would become:

```
?vars.foo.bar=123
```

But this is a bit complicated, because we also allow adding `.` in keys, so we need to quote the whole thing, e.g.

```json
{
  "foo.bar": 123
}
```

becomes:

```
?vars."foo.bar"=123
```

---

Fortunately, there's an easier way! You can simply add the JSON in its entirety to the `vars` query param. For instance, the first example could become:

```
?vars={"foo":{"bar":123}}
```

(with query escaping, of course). So, you can just pass `url.QueryEscape(BUILD_PIPELINE_INSTANCE_VARS)` as the query param as is

Reply:
Cheers! This simplify a lot. Will update.

Reply:
Thanks @aoldershaw !